src/pytest_bdd/scenario.py

```diff
@@ -303,12 +304,29 @@ def scenario_wrapper(request: FixtureRequest, _pytest_bdd_example: dict[str, str
 def collect_example_parametrizations(
     templated_scenario: ScenarioTemplate,
 ) -> list[ParameterSet] | None:
-    if templated_scenario.examples is None:
-        return None
-    if contexts := list(templated_scenario.examples.as_contexts()):
-        return [pytest.param(context, id="-".join(context.values())) for context in contexts]
-    else:
-        return None
+    parametrizations = []
+    has_multiple_examples = len(templated_scenario.examples) > 1
+
+    for example_id, examples in enumerate(templated_scenario.examples):
+        _tags: set = examples.tags or set()
+        example_marks = []
+        if _tags:
+            with warns(PytestUnknownMarkWarning, match=r"Unknown pytest\.mark\.\w+"):
+                example_marks = [pytest.mark.__getattr__(tag) for tag in _tags]
+
+        for context in examples.as_contexts() or [{}]:
+            test_id = (
+                "-".join((str(example_id), *context.values())) if has_multiple_examples else "-".join(context.values())
```

Comment:
why using the `example_id` to define the ID of the test?

Reply:
I started by building on what @olegpidsadnyi had committed in the superseded PR, and this was the approach he had.

Reply:
Yes, I think we should not include the enumeration unless tests fail, can you try without example_id?